featurebyte/service/offline_store_feature_table_manager.py

```diff
@@ -419,10 +429,33 @@ async def _get_required_aggregate_result_tables(
 
         return sorted(set(filtered_table_names))
 
-    async def _get_online_enabled_feature_lists(self) -> List[FeatureListModel]:
+    async def _get_online_enabled_feature_lists(
+        self,
+        feature_list_to_online_enable: Optional[FeatureListModel] = None,
+        feature_list_to_online_disable: Optional[FeatureListModel] = None,
+    ) -> List[FeatureListModel]:
         feature_lists = []
+        fl_has_been_enabled = False
         async for feature_list_dict in self.feature_list_service.iterate_online_enabled_feature_lists_as_dict():
-            feature_lists.append(FeatureListModel(**feature_list_dict))
+            feature_list = FeatureListModel(**feature_list_dict)
+            if (
+                feature_list_to_online_disable
+                and feature_list.id == feature_list_to_online_disable.id
+            ):
+                # skip the feature list to be disabled
+                continue
```

Comment:
We need this check because now the feature list's `enabled` flag is updated at the very last step of the deployment update process, is that correct?

Reply:
Yes, the feature list's attribute is updated at later step in `DeployService`.
```

        if feast_registry:
            await self.feast_integration_service.handle_online_enabled_features(  # this calls offline store ft manager 
                features=features_offline_feature_table_to_update,
                feature_list_to_online_enable=feature_list,
                update_progress=get_ranged_progress_callback(
                    self.task_progress_updater.update_progress, 72, 98
                ),
            )
            await self._update_progress(
                99, f"Updating deployed feature list ({feature_list.name}) ..."
            )
            await self.feast_integration_service.handle_deployed_feature_list(   # feature list attr updated here
                feature_list=feature_list, online_enabled_features=features
            )
```

Reply:
I see, in that case it seems like an existing bug if I understood correctly. Good to fix it 👍

Reply:
Previously we don't rely on the feature list attribute to check whether the Feast integration is enabled or not. Strictly speaking, it is not a bug.